client-reactjs/src/components/application/Jobs/ManualJobDetail.jsx

```diff
@@ -1,121 +1,159 @@
-          })
-          .then((response) => {
-            if(response.ok) {
-              return response.json();
-            }
-            handleError(response);
-          })
-          .then(data => {
-            setSavingData({saving : false, response : ''});
-            message.success('Your response has been recorded')
-          })
-          .catch(error => {
-            console.log(error);
-          }).finally(setTimeout(() =>{
-            window.location.href = "/"
-        }, 2000))
+  //Get Job Details
+  const getJobDetails = async () =>{
+    const jobDetailResponse = await fetch(`/api/job/job_details?app_id=${applicationId}&job_id=${jobId}`, {
+          headers: authHeader(),
+        })
+    if(jobDetailResponse.ok){
+      const data = await jobDetailResponse.json();
+      setJobDetails(data)
+    }else{
+      message.error('Unable to fetch job details');
     }
+  }
+
+  //When user clicks approve or reject btn
+  const handleResponse = async (e) => {
+    await form.validateFields();
+    setShowModal(false);
+    const { action, responseMessage } = form.getFieldsValue();
+    try{
+        const response = await  fetch("/api/job/manualJobResponse", {
+          headers: authHeader(),
+          method: "POST",
+          body: JSON.stringify({
+            jobExecutionId: jobExecutionId,
+            status: action === "approved" ? "completed" : "failed",
+            manualJob_metadata: {
+              response : action,
+              responseMessage,
+              respondedOn: Date.now(),
+            },
+          }),
+        })
 
-    //When user clicks cancel button
-    const handleCancel = () => {
-      window.location.href = "/"
+        if(response.ok){
+          message.success(' Your response has been saved');
+          setTimeout(() => {
+          window.location.href = "/";
+        }, 2000)
+        }else{
+          throw Error('Error occurred while saving your response. Please try again')
+        }
+    }catch(error){
+      message.error(error.message)
     }
+  };
+
+  //When user clicks cancel button
+  const handleCancel = () => {
+   ;
+  };
+
+  //Button styles
+  const actionBtnsStyle = { marginLeft: "10px" };
+
+  //Buttons on tab pane
+  const actions = (
+    <div>
+      <Button style={actionBtnsStyle} type="primary" onClick={() =>{ setShowModal(true);}}>
+        Take Action
+      </Button>
+      <Button style={actionBtnsStyle} type="primary" ghost onClick={handleCancel}>
+        Cancel
+      </Button>
+    </div>
+  );
+  //Job details
+  const jobData = [
+    { label: "Title", value: jobDetails.title },
+    { label: "Name", value: jobDetails.name },
+    { label: "Job Type", value: jobDetails.jobType },
+    { label: "Contact", value: jobDetails.contact },
+    { label: "Created on", value: jobDetails.createdAt },
+  ];
+
+  return (
+    <div>
+      <div className="assetTitle">Job : {jobDetails.name}</div>
+      <Tabs tabBarExtraContent={actions}>
```

Comment:
Better to see a screenshot of this? I am not sure how Tabs are used in this scenario. I was thinking the user flow would be something similar to the below. 

1. User receives an email about manual task
2. Clicks on the link in the email which will take them to a UI with the details of the manual task along with a Text Area named 'Reason/Additional Info' and then two buttons Approve/Reject  OR show a modal dialog when user clicks on Approve/Reject to capture the reason